CMakeLists.txt

```diff
@@ -1,19 +1,57 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.8)
```

Comment:
Ubuntu 16.04 (xenial) comes with cmake 3.5. I think it would be reasonable to still provide support for that.

Reply:
Nope, https://docs.travis-ci.com/user/reference/xenial/

Reply:
Hm, I guess they use a modified image then? https://launchpad.net/ubuntu/+source/cmake

Reply:
Yeah that seems to be travis specific. Xenial should be 3.5. I would argue it's not hard to install the latest version of CMake but if you want to go that route then it would be ideal to use continue using `target_compile_features` listing the features you are using 
https://cmake.org/cmake/help/v3.5/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html

The less ideal version is to do what's listed here
https://cliutils.gitlab.io/modern-cmake/chapters/features/cpp11.html#cmake-31-global-and-property-settings

Reply:
To keep out of the box Xenial support:
Unless I missed one - https://godbolt.org/z/9JmA7Q
`cxx_std_11` can be replaced with
```
cxx_auto_type 
cxx_constexpr
cxx_decltype 
cxx_deleted_functions 
cxx_generalized_initializers 
cxx_nullptr 
cxx_right_angle_brackets
cxx_rvalue_references 
cxx_strong_enums
cxx_variable_templates
```
Which should actually be better as CMake should be able to tell whether the compiler supports that C++11 feature instead of a blanket "does the compiler have a C++11 mode".

Reply:
Which ones exactly do we use in thos project?

Reply:
You don't need to have either `project` or `cmake_minimum_required` in these CMakeLists. Unless you have any reason for them to work without the top level CMakeLists?

Reply:
You are correct!

Reply:
Done!